athena-gcs/src/main/java/com/amazonaws/athena/connectors/gcs/filter/FilterExpressionBuilder.java

```diff
@@ -0,0 +1,96 @@
+import com.amazonaws.athena.connector.lambda.domain.predicate.ValueSet;
+import org.apache.arrow.vector.types.pojo.Field;
+import org.apache.arrow.vector.types.pojo.Schema;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+/**
+ * This class is used to build Filter Expression to handle the constraints on partition folder
+ *
+ */
+public class FilterExpressionBuilder
+{
+    private static final Logger LOGGER = LoggerFactory.getLogger(FilterExpressionBuilder.class);
+
+    /**
+     * List of {@link Field} instances
+     */
+    private final List<Field> fields;
+
+    public FilterExpressionBuilder(Schema schema)
+    {
+        this.fields = schema.getFields();
+    }
+
+    /**
+     * Prepares the expressions based on the provided arguments.
+     *
+     * @param constraints            An instance of {@link Constraints} that is a summary of where clauses (if any)
+     * @return A list of {@link EqualsExpression}
+     */
+    public List<AbstractExpression> getExpressions(Constraints constraints)
+    {
+        LOGGER.info("Constraint summaries: \n{}", constraints.getSummary());
+        List<AbstractExpression> conjuncts = new ArrayList<>();
+        for (Field column : fields) {
+            if (constraints.getSummary() != null && !constraints.getSummary().isEmpty()) {
+                ValueSet valueSet = constraints.getSummary().get(column.getName());
+                LOGGER.info("Value set for column {} was {}", column, valueSet);
+                if (valueSet != null) {
+                    conjuncts.add(getFilterExpressions(column.getName(), valueSet));
+                }
+            }
+        }
+        return conjuncts;
+    }
+
+    /**
+     * Returns an {@link AbstractExpression} based on {@link ValueSet}
+     *
+     * @param columnName The name of the column
+     * @param valueSet   An instance of {@link ValueSet}
+     * @return {@link AbstractExpression}
+     */
+    private AbstractExpression getFilterExpressions(String columnName, ValueSet valueSet)
+    {
+        LOGGER.info("FilterExpressionBuilder::getFilterExpressions -> Evaluating and adding expression for col {} with valueSet {}", columnName, valueSet);
+        LOGGER.info("FilterExpressionBuilder::getFilterExpressions -> Bound {}", valueSet.getRanges().getSpan().getLow().getBound());
+
+        List<Object> singleValues = valueSet.getRanges().getOrderedRanges().stream()
+          .filter(range -> range.isSingleValue())
+          .map(range -> range.getLow().getValue())
+          .collect(Collectors.toList());
+
+        if (singleValues.size() == 1) {
+            return new EqualsExpression(columnName.toLowerCase(), singleValues.get(0));
+        }
+
+        return new AnyExpression(columnName.toLowerCase(), singleValues.stream().map(Object::toString).collect(Collectors.toList()));
```

Comment:
It looks like `AnyExpression` is relying on `FilterExpressionBuilder::getFilterExpressions` to convert everything to Strings so that it can work, I'm not sure if this is actually correct.

Reply:
we support only string constraints for partition columns and we are dealing with only partition constraits here.

Reply:
Sure but if we are only dealing with strings then we could just cast the objects to strings instead of doing `toString()`. When I try casting instead, it will result in an error because the underlying types are not strings, but do have `toString()` methods available.

Reply:
var SingleValues is of class org.apache.arrow.vector.util.Text . and we can't directly cast it to String as there is a toString() method available. (https://arrow.apache.org/docs/java/reference/org/apache/arrow/vector/util/Text.html).

I tried casting, but it threw an error: class org.apache.arrow.vector.util.Text cannot be cast to class java.lang.String.

Reply:
Right, so that's my point, that its not actual Strings, but some other class.

I'm going to try to refactor this code as well and simplify things.

Reply:
Here's my change to simplify things: https://github.com/trianz-azam/aws-athena-query-federation/commit/31f24536d14b0dba7d20018ec121214102fd6b04

Please verify that this works (as in run your integration tests and manual testing to verify that the connector continues to work as before).

Reply:
yes, It is working as expected.